sklearn/base.py

```diff
@@ -570,6 +573,12 @@ def _validate_data(
         no_val_y = y is None or isinstance(y, str) and y == "no_validation"
 
         default_check_params = {"estimator": self}
+        if hasattr(self, "warm_start") and self.warm_start:
+            try:
+                check_is_fitted(self)
+                default_check_params["force_all_finite"] = False
+            except Exception:
+                pass
```

Comment:
Checking for `warm_start` in `BaseEstimator` is too estimator specific and would change behavior of third party estimators that inherit from `BaseEstimator`. 

I think the logic should be done in the child class.

Reply:
Should I revert this change and others made in other modules (as you suggested in your original comment) ? 

I started doing it in the child classes but it was quite long to change all the estimators that had a warm start option and i thought it would be better for both future addings and current implementation to do it in `BaseEstimator`.

Reply:
> Should I revert this change and others made in other modules (as you suggested in your original comment) ?

Yes. Changing `BaseEstimator` drastically increases the scope of the PR, because it effects all estimators. Making the PR smaller makes it easier to review. This is especially important when the core-devs did not decide on the way forward. A smaller PR means that the context is easier to understand and which makes it easier to decide on.

Once this PR is merged, we can have follow up PRs that add the feature to other estimators. At that point, we can decide on refactoring the logic into `BaseEstimator`.

Reply:
I reverted the changes in other files and kept the one in `sklearn.ensemble._gb.py`. I'm waiting for the pipelines to finish but i think it's good now